src/EditorFeatures/Core/Implementation/CommentSelection/ToggleLineCommentCommandHandler.cs

```diff
@@ -141,8 +157,40 @@ public bool ExecuteCommand(ToggleLineCommentCommandArgs args, CommandExecutionCo
                 }
             }
 
-            trackingSpans.Add(new CommentTrackingSpan(
-                TextSpan.FromBounds(linesInSelection.First().Start, linesInSelection.Last().End)));
+            var commentTrackingSpan = GetNewSelection(selectedSpan, editorOptions, isMultiCaret);
+            trackingSpans.Add(commentTrackingSpan);
+        }
+
+        private static CommentTrackingSpan GetNewSelection(SnapshotSpan selectedSpan, IEditorOptions editorOptions, bool isMultiCaret)
+        {
+            // If the span is just the caret location without a selection move the caret
+            // down to the next line at the same column (or end of line).
+            // We bail in multi caret scenarios as the next line could already be modified as part of this comment operation.
+            if (selectedSpan.IsEmpty && !isMultiCaret)
+            {
+                selectedSpan.End.GetLineAndCharacter(out var caretLineNumber, out var caretOffset);
+                var caretLine = selectedSpan.Snapshot.GetLineFromLineNumber(caretLineNumber);
+
+                var nextLineNumber = caretLineNumber + 1;
+                if (nextLineNumber < selectedSpan.Snapshot.LineCount)
+                {
+                    // Get the column in the line of the current caret location.
+                    var caretColumn = caretLine.GetColumnFromLineOffset(caretOffset, editorOptions);
+
+                    var nextLine = selectedSpan.Snapshot.GetLineFromLineNumber(nextLineNumber);
+                    // Compute the correct offset location from the column in the previous line.
+                    var offsetInNextLine = nextLine.GetLineOffsetFromColumn(caretColumn, editorOptions);
+                    var position = selectedSpan.Snapshot.GetPosition(nextLineNumber, offsetInNextLine);
+                    return new CommentTrackingSpan(TextSpan.FromBounds(position, position));
```

Comment:
does all this code work if the position in the line you're on goes *past* the end of the next line?  if so, do we have test?  thanks!

Reply:
@CyrusNajmabadi Yeah - GetLineOffsetFromColumn will just return the end of the line if the column is outside the bounds of the line

For a test, see this one - https://github.com/dotnet/roslyn/pull/51411/files#diff-01321fe56eaeeda7a606332cae99ec00cddea81211ebf7a8196a2100ea0d9a0cR59
The caret is after `var` which is in a column past the end of the next line.  The caret then just moves to the end of the next line.

Reply:
terrific.  thanks!  might be worth a comment if you're editing htis method.